homeassistant/components/dunehd/media_player.py

```diff
@@ -38,33 +37,42 @@
     SUPPORT_PAUSE
     | SUPPORT_TURN_ON
     | SUPPORT_TURN_OFF
-    | SUPPORT_SELECT_SOURCE
     | SUPPORT_PREVIOUS_TRACK
     | SUPPORT_NEXT_TRACK
     | SUPPORT_PLAY
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the DuneHD media player platform."""
-
-    sources = config.get(CONF_SOURCES, {})
+    """Set up the Dune HD media player platform."""
     host = config.get(CONF_HOST)
-    name = config.get(CONF_NAME)
 
-    add_entities([DuneHDPlayerEntity(DuneHDPlayer(host), name, sources)], True)
+    hass.async_create_task(
```

Comment:
This is using the async api from sync context. We need to change to use the coroutine `async_setup_platform`.

Reply:
Sorry about that. I'll fix it in a moment.

Reply:
No worries, thanks!

Reply:
Done https://github.com/home-assistant/core/pull/36408